llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -10867,6 +10867,16 @@ SDValue TargetLowering::expandAddSubSat(SDNode *Node, SelectionDAG &DAG) const {
   assert(VT == RHS.getValueType() && "Expected operands to be the same type");
   assert(VT.isInteger() && "Expected operands to be integers");
 
+  // usub.sat(a, 1) -> sub(a, zext(a != 0))
+  if (Opcode == ISD::USUBSAT && isOneConstant(RHS)) {
```

Comment:
isOneConstant only returns true for scalar. So this still doesn't support vector.